torch/csrc/Module.cpp

```diff
@@ -2105,6 +2106,13 @@ Call this whenever a new thread is created in order to propagate values from
       "_set_conj", [](const at::Tensor& x, bool conj) { x._set_conj(conj); });
   py_module.def(
       "_set_neg", [](const at::Tensor& x, bool neg) { x._set_neg(neg); });
+  py_module.def(
+      "_set_storage",
```

Comment:
(trying to bikeshed on the name.... `_unsafe_set_storage`/`_set_only_storage`?)

@albanD mostly wondering if you are unhappy with a dedicated API for this. Basically, `x.set_(storage, size, stride)`: does 3 things right now:

(1) set the storage
(2) call `set_sizes_strides()
(3) potentially resize the storage (if the number of bytes required for the given size/stride is larger than the nbytes of the passed-in storage

And `return_and_correct_aliasing` (used by subclasses) only ever calls `a.set_(b.storage(), a.size(), a.stride())`, where we are guaranteed that the resize and the `set_sizes_strides_` are not necessary.

Those extra two steps are (1) a pain for nested tensor (trying to compute nbytes), and (2) unnecessary work we are doing throughout tracing (since this is used by `FunctionalTensor` as well)

Reply:
Agreed, your suggested names are much better - do we not already have something like this? Just looked and there's `_unsafe_reset_storage()` that seems to be a functionalization thing.